api/src/glfs-fops.c

```diff
@@ -665,11 +665,14 @@ pub_glfs_openat(struct glfs_fd *pglfd, const char *path, int flags, mode_t mode)
     if (!is_create)
         ret = syncop_open(subvol, &loc, flags, glfd->fd, fop_attr, NULL);
     else
-        ret = syncop_create(subvol, &loc, flags, mode, glfd->fd, NULL, fop_attr,
-                            NULL);
+        ret = syncop_create(subvol, &loc, flags, mode, glfd->fd, &iatt,
+                            fop_attr, NULL);
 
     DECODE_SYNCOP_ERR(ret);
 
+    if (ret == 0)
```

Comment:
Shouldn't we also check that it's a create fop ? I would say we don't need to link the inode for an open fop.

Reply:
You are correct. I'll update shortly.